modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java

```diff
@@ -164,265 +164,139 @@ long tryGetStreamLength(GetObjectResponse getObjectResponse) {
             if (rangeEnd < rangeStart) {
                 throw new IllegalArgumentException("invalid Content-range header [" + rangeString + "]");
             }
-            if (rangeStart != start + currentOffset) {
+            if (rangeStart != expectedStart) {
                 throw new IllegalArgumentException(
-                    "unexpected Content-range header [" + rangeString + "], should have started at " + (start + currentOffset)
+                    "unexpected Content-range header [" + rangeString + "], should have started at " + expectedStart
                 );
             }
-            if (rangeEnd > end) {
+            if (rangeEnd > expectedEnd) {
                 throw new IllegalArgumentException(
-                    "unexpected Content-range header [" + rangeString + "], should have ended no later than " + end
+                    "unexpected Content-range header [" + rangeString + "], should have ended no later than " + expectedEnd
                 );
             }
             return rangeEnd - rangeStart + 1L;
         }
         return getObjectResponse.contentLength();
     }
 
-    @Override
-    public int read() throws IOException {
-        ensureOpen();
-        final int initialAttempt = attempt;
-        while (true) {
-            try {
-                final int result = currentStream.read();
-                if (result == -1) {
-                    eof = true;
-                } else {
-                    currentOffset += 1;
-                }
-                maybeLogAndRecordMetricsForSuccess(initialAttempt, "read");
-                return result;
-            } catch (IOException e) {
-                if (attempt == initialAttempt) {
-                    blobStore.getS3RepositoriesMetrics().retryStartedCounter().incrementBy(1, metricAttributes("read"));
-                }
-                reopenStreamOrFail(e);
-            }
+    private static class S3InputStream extends InputStream {
```

Comment:
This name feels too generic. Can we rename it to something more meaningful to the context, e.g. somethig like `S3CurrentOpenStream`?

Reply:
Done in 55c48862c34091bbc633369ae97bdd44f780f96f